pkg/fission-cli/httptrigger.go

```diff
@@ -59,11 +59,12 @@ func getMethod(method string) string {
 	return ""
 }
 
-func setHtFunctionRef(functionList []string, functionWeightsList []int) (*fv1.FunctionReference, error) {
+func setHtFunctionRef(functionList []string, functionWeightsList []int, functionTimeout uint64) (*fv1.FunctionReference, error) {
 	if len(functionList) == 1 {
 		return &fv1.FunctionReference{
-			Type: fv1.FunctionReferenceTypeFunctionName,
-			Name: functionList[0],
+			Type:            fv1.FunctionReferenceTypeFunctionName,
+			Name:            functionList[0],
+			FunctionTimeout: functionTimeout,
```

Comment:
FunctionReference seems an odd place for adding a timeout attribute - reference should be that's it - a reference. I haven't looked at whole flow - but I suggest revisiting the decision of placing timeout in reference struct!